icu4c/source/i18n/usearch.cpp

```diff
@@ -3668,6 +3689,16 @@ static int32_t nextBoundaryAfter(UStringSearch *strsrch, int32_t startIndex) {
     UBreakIterator *breakiterator = strsrch->search->breakIter;
 
     if (breakiterator == NULL) {
+        // Lazily create the internal break iterator.
```

Comment:
It may make sense to put this creation logic in a little `getInternalBreakIterator()` function, to avoid repeating it multiple times. Or maybe it could be `getBreakIterator()`, and include the test for a user supplied one.

Reply:
Thanks Andy! That's a good idea!
I think I'll add your suggested `getBreakIterator()` function, and then replace the repeated/redundant code with it.